models/DataObject/ClassDefinition/Service.php

```diff
@@ -358,4 +360,63 @@ public static function skipColumn($tableDefinitions, $table, $colName, $type, $d
 
         return false;
     }
+
+    /**
+     * @param array $implementsParts
+     * @param string $newInterfaces
+     * @return string
+     * @throws \Exception
+     */
+    public static function buildImplementsInterfaces($implementsParts = [], string $newInterfaces) {
+
+        if ($newInterfaces) {
+            $customParts = explode(',', $newInterfaces);
+            foreach ($customParts as $interface) {
+                $interface = trim($interface);
+                if (Tool::interfaceExists($interface)) {
+                    $implementsParts[]= $interface;
+                } else {
+                    throw new \Exception("interface '" . $interface . "' does not exist");
+                }
+            }
+        }
+
+        if ($implementsParts) {
+            return ' implements ' . implode(', ', $implementsParts);
+        }
+        return '';
+    }
+
+
+    /**
+     * @param array $useParts
+     * @param string $newTraits
+     * @return string
+     * @throws \Exception
+     */
+    public static function buildUseTraits($useParts = [], string $newTraits) {
+
+        if ($newTraits) {
+            $customParts = explode(',', $newTraits);
+            foreach ($customParts as $trait) {
+                $trait = trim($trait);
+                if (Tool::classInterfaceExists($trait, 'trait')) {
+                    $useParts[]= $trait;
+                } else {
+                    throw new \Exception("trait '" . $trait . "' does not exist");
+                }
+            }
+        }
+
+        if ($useParts) {
+            $result = '';
+            foreach ($useParts as $part) {
+                $result .= 'use ' . $part . ";\r\n";
```

Comment:
Everywhere else, only `\n` is used.
```suggestion
                $result .= 'use ' . $part . ";\n";
```